hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroSource.java

```diff
@@ -32,4 +33,9 @@ public AvroSource(TypedProperties props, JavaSparkContext sparkContext, SparkSes
       SchemaProvider schemaProvider) {
     super(props, sparkContext, sparkSession, schemaProvider, SourceType.AVRO);
   }
+
+  @Override
+  protected final InputBatch<JavaRDD<GenericRecord>> fetchNewData(Option<String> lastCkptStr, long sourceLimit) {
```

Comment:
this is an abstract class. so why the need to implement this - vs forcing subclasses to deal with it